hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/Task.java

```diff
@@ -10,10 +11,25 @@
   private Protos.TaskInfo taskInfo;
   private String cmd;
   private Process process;
+  private String type;
 
   public Task(Protos.TaskInfo taskInfo) {
     this.taskInfo = taskInfo;
     this.cmd = taskInfo.getData().toStringUtf8();
+    discoveryType(taskInfo.getTaskId().getValue());
+  }
+
```

Comment:
We will only be setting the type in this class. discoveryType should be a private setType method. We should remove the public setType method since no one should call that.